components/accelerator/nvidia/infiniband/component_output.go

```diff
@@ -94,52 +92,33 @@ func ParseStatesToOutput(states ...components.State) (*Output, error) {
 	return nil, errors.New("no state found")
 }
 
+var msgNoAtLeastPortsOrRateSet = "no at least ports or at least rate set"
+
 // Returns the output evaluation reason and its healthy-ness.
+// We DO NOT auto-detect infiniband devices/PCI buses, strictly rely on the user-specified config.
 func (o *Output) Evaluate(cfg Config) (string, bool, error) {
+	// nothing specified for this machine, gpud MUST skip the ib check
+	if cfg.AtLeastPorts == 0 && cfg.AtLeastRate == 0 {
+		return msgNoAtLeastPortsOrRateSet, true, nil
+	}
+
 	if !infiniband.SupportsInfinibandProduct(o.GPUProductName) {
 		return fmt.Sprintf("%q GPUs do not support infiniband", o.GPUProductName), true, nil
 	}
 
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-	defer cancel()
-	count, err := infiniband.CountInfinibandPCIBuses(ctx)
-	if err != nil {
-		return fmt.Sprintf("failed to count infiniband pci buses: %s", err), false, nil
-	}
-	if count == 0 {
-		return "no infiniband pci buses found", true, nil
+	if len(o.Ibstat.Errors) > 0 {
+		return fmt.Sprintf("ibstat errors found: %s", strings.Join(o.Ibstat.Errors, ", ")), false, nil
 	}
 
-	if o.InfinibandClassExists && o.IbstatExists {
-		if len(o.Ibstat.Errors) > 0 {
-			return fmt.Sprintf("infiniband suppported but ibstat errors found: %s", strings.Join(o.Ibstat.Errors, ", ")), false, nil
-		}
-		if len(o.Ibstat.Parsed) > 0 {
-			// Get expected port count
-			atLeastPorts := cfg.ExpectedPortStates.AtLeastPorts
-
-			// some H100 machines only have 1 ib port in ib class dir
-			if atLeastPorts == 0 {
-				atLeastPorts = infiniband.CountInfinibandClass()
-			}
-
-			// H100 machines with 12 ib ports should default to the GPU count 8
-			if atLeastPorts == 0 || atLeastPorts > o.GPUCount {
-				atLeastPorts = o.GPUCount
-			}
-
-			// no rate is set, use the default rate based on the product
-			atLeastRate := cfg.ExpectedPortStates.AtLeastRate
-			if atLeastRate == 0 {
-				atLeastRate = infiniband.SupportsInfinibandPortRate(o.GPUProductName)
-			}
-
-			if err := o.Ibstat.Parsed.CheckPortsAndRate(atLeastPorts, atLeastRate); err != nil {
-				return err.Error(), false, nil
-			}
+	if len(o.Ibstat.Parsed) > 0 {
```

Comment:
Why return "no ibstat issue/error found" when `o.Ibstat.Parsed` is empty?

I think if the node group expects the node to have IB devices, then get empty parsed message should also be an error type?

Reply:
Good catch, fixing.

Reply:
@FillZpp Addressed, PTAL.